lib/puppet/type/jenkins_credentials.rb

```diff
@@ -51,26 +51,152 @@
 
   newproperty(:password) do
     desc 'password - UsernamePasswordCredentialsImpl, CertificateCredentialsImpl'
+    def is_to_s(_current_value)
+      '[old password]'
+    end
+
+    def should_to_s(_new_value)
+      '[new password]'
+    end
+
+    def change_to_s(_current_value, _new_value)
+      begin
+        if _current_value == :absent
+          return "Added '#{name}' as #{should_to_s(_new_value)}"
+        elsif _new_value == :absent or _new_value == [:absent]
+          return "Removed '#{name}' from #{is_to_s(_current_value)}"
+        else
+          return "Changed '#{name}' from #{is_to_s(_current_value)} to #{should_to_s(_new_value)}"
+        end
+      rescue Puppet::Error
+        raise
```

Comment:
What's the point of this?

Also, since Ruby 2.5 you can write:
```ruby
def my_method
  # ...
rescue Puppet::Error
  # ...
end
```

Reply:
This was in the code i copied. Clearly unneeded so i removed it.